### PR TITLE
[MOB-5759] processor

### PR DESCRIPTION
**Pull request by roninopf**
## 🔹 Jira Ticket(s)

* [MOB-5759](https://iterable.atlassian.net/browse/MOB-5759)

## ✏️ Description

* adds a pure struct/function message processor

[MOB-5759]: https://iterable.atlassian.net/browse/MOB-5759?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: swift-sdk/Internal/EmbeddedMessagingManager.swift [hunk 1]
```diff
@@ -133,5 +148,5 @@ class EmbeddedMessagingManager: NSObject, IterableEmbeddedMessagingManagerProtoc
     
     private var messages: [IterableEmbeddedMessage] = []
     
-    private var listeners: [String] = [] //change to protocol
+    private var listeners: [IterableEmbeddedMessagingUpdateDelegate] = []
 }
```
Comment (evangreer91) [inline swift-sdk/Internal/EmbeddedMessagingManager.swift:152]: hm interesting so all the logic is passed to the processor and the manager just directs calls basically.

File: swift-sdk/Internal/EmbeddedMessagingProcessor.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    func placementIdsToNotify() -> [String] {
+        return getNewMessages()
+            .map { $0.metadata.placementId }
+    }
+    
+    private let currentMessages: [IterableEmbeddedMessage]
+    private let fetchedMessages: [IterableEmbeddedMessage]
+
```
Comment (evangreer91) [inline swift-sdk/Internal/EmbeddedMessagingProcessor.swift:31]: Should currentMessages and fetchedMessages be moved to the top?

Comment (roninopf) [reply swift-sdk/Internal/EmbeddedMessagingProcessor.swift:31]: No, on iOS we keep the private variables and functions at the bottom.

File: swift-sdk/Internal/EmbeddedMessagingManager.swift (hunk 1 above)
Comment (roninopf) [reply swift-sdk/Internal/EmbeddedMessagingManager.swift:152]: More or less, it's about keeping the details separate from the manager.

File: swift-sdk/IterableEmbeddedMessagingUpdateDelegate.swift [hunk 3]
```diff
@@ -0,0 +1,9 @@
+//
+//  Copyright © 2023 Iterable. All rights reserved.
+//
+
+import Foundation
+
+@objc public protocol IterableEmbeddedMessagingUpdateDelegate {
+    func onMessagesUpdated()
+}
```
Comment (evangreer91) [inline swift-sdk/IterableEmbeddedMessagingUpdateDelegate.swift:9]: assuming this is added on another PR

Comment (roninopf) [reply swift-sdk/IterableEmbeddedMessagingUpdateDelegate.swift:9]: Yeah, I had started on implementing update delegates, but then I saw a need to implement the processor first.

**Pull request merged**